devito/mpi/halo_scheme.py

```diff
@@ -39,54 +40,53 @@ class HaloScheme(object):
 
     Where ``HaloSchemeEntry`` is a (named) 2-tuple:
 
-        ``({loc_indices}, ((Dimension, DataSide, amount), ...))``
+        ``(loc_indices={}, halos=[(Dimension, DataSide), ...])``
 
-    The tuples (Dimension, DataSide, amount) tell the amount of data that
-    a DiscreteFunction should communicate along its Dimensions.
+    ``loc_indices`` is a dict telling how to access/insert the halo along non-halo
+    indices. For example, consider the Function ``u(t, x, y)``. Assume ``x`` and
+    ``y`` require a halo exchange. The question is: once the halo exchange is
+    performed, at what offset in ``t`` should it be placed? should it be at ``u(0,
+    ...)`` or ``u(1, ...)`` or even ``u(t-1, ...)``? ``loc_indices`` has as many
+    entries as non-halo Dimensions, and each entry provides symbolic information
+    about how to access the corresponding non-halo Dimension. For example, here
+    ``loc_indices`` could be ``{t: t-1}``.
 
-    The dict ``loc_indices`` tells how to access/insert the halo along the
-    keyed Function's non-halo indices. For example, consider the Function
-    ``u(t, x, y)``. Assume ``x`` and ``y`` require a halo exchange. The
-    question is: once the halo exchange is performed, at what offset in ``t``
-    should it be placed? should it be at ``u(0, ...)`` or ``u(1, ...)`` or even
-    ``u(t-1, ...)``? ``loc_indices`` has as many entries as non-halo
-    dimensions, and each entry provides symbolic information about how to
-    access the corresponding non-halo dimension. Thus, in this example
-    ``loc_indices`` could be, for instance, ``{t: 0}`` or ``{t: t-1}``.
+    ``halos`` is a list of 2-tuples ``(Dimension, DataSide)``. This is metadata
```

Comment:
That looks like a tuple of 2-tuples?

Reply:
yes...unless I see other major (coding-related) issues, I'll fix docstrings in the next (upcoming) PR